sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -203,6 +203,17 @@ def test_kernel_pca_n_components():
             assert shape == (2, c)
 
 
+@pytest.mark.parametrize("solver", ["dense", "arpack", "randomized"])
```

Comment:
This parametrization is not necessary. The validation of ``n_components`` does not depend on the solver

Reply:
Wasn't sure if it was worth keeping in there, e.g. just in case in the future its validation was moved elsewhere. Removed now, thanks!

Reply:
Unlikely since param validation should happen at the very beginning of fit